server/services/objectAssetService.js

```diff
@@ -62,28 +66,69 @@ async function getAssetsForArtworks(artworks) {
   // Otherwise, we're fetching a single artwork object's renditions
   // so we do need archival renditions as part of our list
   // and need some extra work to do so
-  const { objectNumber } = artworksInformation[0];
+  const artwork = { ...artworks[0] };
+  const artworkWithDAMSInformation = await addAssetFields(artwork);
+
+  return [artworkWithDAMSInformation];
+}
+
+async function getEnsembleImageUrl(ensembleIndex) {
+  const ensembleCacheKey = makeEnsembleCacheKey(ensembleIndex);
+  const ensembleImageUrl = memoryCache.get(ensembleCacheKey);
+
+  // If we don't have a cached version of this ensemble image url
+  // let's fetch it live, store it, then return it
+  if (!ensembleImageUrl) {
+    const liveEnsembleImageUrl = await damsService.getEnsembleImageUrl(
+      ensembleIndex
+    );
+    memoryCache.put(ensembleCacheKey, liveEnsembleImageUrl, oneWeek);
+
+    return liveEnsembleImageUrl;
+  }
+
+  // Otherwise, we have the cached version
+  return ensembleImageUrl;
+}
```

Comment:
Utilizes the `damsService.getEnsembleImageUrl` function to get an asset URL for a provided ensemble index. 

It caches it to the server to prevent subsequent fetches from NetX